test/files/run/t10513.scala

```diff
@@ -0,0 +1,35 @@
+//package scala.concurrent
+import scala.concurrent._
+import duration._
+
+import scala.util.{Random, Try}
+import ExecutionContext.Implicits.global
+
+/** This test uses recursive calls to Future.flatMap to create arrays whose
+ *  combined size is slightly greater than the JVM heap size. A previous
+ *  implementation of Future.flatMap would retain references to each array,
+ *  resulting in a speedy OutOfMemoryError. Now, each array should be freed soon
+ *  after it is created and the test should complete without problems.
+ */
+object Test {
+ val arrSz = 50 * 10000
```

Comment:
Please make sure to indent this with 2 spaces.

Reply:
Thanks I was also irked. I think those vals are scheduled to be moved anyway. But it makes me miss a formatting option, scalafmt etc. It's the little things...